extractor/text.go

```diff
@@ -1180,17 +1181,46 @@ var (
 // sortPosition sorts a text list by its elements' positions on a page.
 // Sorting is by orientation then top to bottom, left to right when page is orientated so that text
 // is horizontal.
+// Text is considered to be on different lines if the lines' orientedStart.Y differs by more than `tol`.
 func (pt *PageText) sortPosition(tol float64) {
+	if len(pt.marks) == 0 {
+		return
+	}
 	sort.SliceStable(pt.marks, func(i, j int) bool {
 		ti, tj := pt.marks[i], pt.marks[j]
 		if ti.orient != tj.orient {
 			return ti.orient < tj.orient
 		}
-		if math.Abs(ti.orientedStart.Y-tj.orientedStart.Y) > tol {
+		if ti.orientedStart.Y != tj.orientedStart.Y {
 			return ti.orientedStart.Y > tj.orientedStart.Y
 		}
 		return ti.orientedStart.X < tj.orientedStart.X
 	})
+
+	// Compute yOrder. yOrder increments if orientedStart.Y differs by more than `tol`.
```

Comment:
This seems more complicated than the previous algorithm.  What is the reason for added complexity?
Seems like a single sort function should be enough if the condition is only:
```
Sorting is by orientation then top to bottom, left to right when page is orientated so that text	// Sorting is by orientation then top to bottom, left to right when page is orientated so that text is horizontal.
```
In that case the less function should simply be
```
  if yi == yj {
    return xi < xj
  }
  return yi >= yj
```
To account for the `tol`erance then  I think the previous function would have worked
except should use 
```
if math.Abs(ti.orientedStart.Y-tj.orientedStart.Y) > tol {		
			return ti.orientedStart.Y >= tj.orientedStart.Y
}
return ti.orientedStart.X < tj.orientedStart.X
```

I don't quite see why this should not work.

Reply:
I think I need to sort twice because the desired order depends on the difference between the Y coordinates of text marks, not just the Y coordinates of individual text marks.